### DISCUSSION TITLE
How to make Dynamic Parallel Queries that depend on another query ?

### DISCUSSION THREAD

**Discussion by rossifumi46** [TanStack/query#2323] [Q&A]
Something like this:  
```
const userQueries = useQueries(
    parkings.map((parking) => {
      return {
        queryKey: ['getDays', { parkingPlaceId: parking },  token]
        queryFn: getDays(),
      }
    }), {
      enabled: !!parkings,
    }
  )
```

**Comment by TkDodo** [comment]
If they depend on the previous one, you can’t run them in parallel 🙃

**Reply by rossifumi46** [reply]
sorry for my english. I mean they all depend on another query that returns parkings list

**Comment by TkDodo** [comment]
Ah, like so?

```
const userQueries = useQueries(
    parkings.map((parking) => {
      return {
        queryKey: ['getDays', { parkingPlaceId: parking },  token]
        queryFn: getDays(),
        enabled: !!parkings,
      }
    }),
  )
```

**Reply by onkarj422** [reply]
But, if parkings list is empty, it will mount the useQueries with an empty array. And it doesn't seem to be working that way because even when parkings list is updated, actual useQueries hook won't update

**Reply by TkDodo** [reply]
Can you create a codesandbox example of that behavior?
